projects/packages/assets/src/class-assets.php

```diff
@@ -321,15 +334,26 @@ public static function normalize_path( $path ) {
 	 *  - `media`:            (string) Media for the css file. Default 'all'.
 	 *  - `minify`:           (bool|null) Set true to pass `minify=true` in the query string, or `null` to suppress the normal `minify=false`.
 	 *  - `nonmin_path`:      (string) Non-minified script path.
+	 *  - `strategy`:         (string) Specify a script strategy to use, eg. `defer` or `async`. Default is `""`.
 	 *  - `textdomain`:       (string) Text domain for the script. Required if the script depends on wp-i18n.
 	 *  - `version`:          (string) Override the version from the `asset_path` file.
 	 * @throws \InvalidArgumentException If arguments are invalid.
 	 */
 	public static function register_script( $handle, $path, $relative_to, array $options = array() ) {
+		global $wp_version;
+
 		if ( substr( $path, -3 ) !== '.js' ) {
 			throw new \InvalidArgumentException( '$path must end in ".js"' );
 		}
 
+		if ( $options['async'] ) {
+			_deprecated_argument(
+				__METHOD__,
+				'$$next-version$$',
+				'The `async` option is deprecated in favor of `strategy`'
```

Comment:
May as well be clear here too.
```suggestion
				'The `async` option is deprecated in favor of `strategy` set to "defer"'
```